api/lib/src/lp.rs

```diff
@@ -17,6 +17,31 @@ use state_chain_runtime::RuntimeCall;
 use std::ops::Range;
 use utilities::rpc::NumberOrHex;
 
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub enum ApiWaitForResult<T> {
+	TransactionHash(H256),
+	Details { tx_hash: H256, response: T },
```

Comment:
Also I think maybe different variant names, as they will show up in the jsonrpc response would be good, maybe "TxHash" and "TxResult"